frontend/libs/studio-components/src/components/StudioSpinner/StudioSpinner.test.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { StudioSpinner } from './StudioSpinner';
+import { textMock } from '../../../../../testing/mocks/i18nMock';
+
+const mockSpinnerText: string = 'Test text';
+const mockTestId: string = 'testId';
+
+jest.mock('react', () => ({
+  ...jest.requireActual('react'),
+  useId: () => mockTestId,
+}));
+
+describe('StudioSpinner', () => {
+  afterEach(jest.clearAllMocks);
+
+  it('should render the spinner without spinnerText when it is not present', () => {
+    render(<StudioSpinner />);
+
+    const spinnerTitle = screen.getByText(textMock('general.loading'));
+    expect(spinnerTitle).toBeInTheDocument();
```

Comment:
What do you think about simplifying and using the `getByTitle` instead to ensure the text is accessible?
```suggestion
    expect(screen.getByTitle(textMock('general.loading'))
```